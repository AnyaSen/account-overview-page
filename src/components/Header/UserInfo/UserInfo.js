import React from "react";

import Styles from "./UserInfo.module.scss";

export default function UserInfo({ userName, userPicture, accountsNumber }) {
  return (
    <div className={Styles.UserInfoContainer}>
      <div className={Styles.UserImgContainer}>
        <img src={userPicture} alt="User Face" />
        <p>{userName}</p>
      </div>

      <p>Number of accounts: {accountsNumber}</p>
    </div>
  );
}
