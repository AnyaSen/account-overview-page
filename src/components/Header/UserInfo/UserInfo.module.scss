@import "../../../assets/scss/variables";

.UserInfoContainer {
  display: flex;
  justify-content: space-between;
  padding: $space-2 $space-2;
}

.UserImgContainer {
  display: flex;
  align-items: center;

  img {
    height: $space-4;
    width: auto;
    border-radius: 2rem;
    box-shadow: $shadow-1;
    align-self: flex-end;
    margin-right: $space-1;
  }
}
