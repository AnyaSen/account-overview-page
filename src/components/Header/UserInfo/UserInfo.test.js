import React from "react";
import { render, cleanup } from "@testing-library/react";

import UserInfo from "./UserInfo";

describe("<UserInfo />", () => {
  afterAll(cleanup);

  test("Should have correct name & number.", () => {
    const name = "John Smith";
    const accNumber = 70;
    const userPic = "userPic.svg";

    const { getByTestId, getByAltText, container, getByText } = render(
      <UserInfo
        userName={name}
        accountsNumber={accNumber}
        userPicture={userPic}
      />
    );

    const UserNumber = getByTestId("number-of-accounts");
    const UserText = getByText(name);
    const userPicture = getByAltText("User Face");

    expect(UserNumber).toBeTruthy();
    expect(UserText).toBeTruthy();
    expect(userPicture).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
