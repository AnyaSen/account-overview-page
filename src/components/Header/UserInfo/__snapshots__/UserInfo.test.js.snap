// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInfo /> Should have correct name & number. 1`] = `
<div
  class="UserInfoContainer"
>
  <div
    class="UserImgContainer"
  >
    <img
      alt="User Face"
      src="userPic.svg"
    />
    <p>
      John Smith
    </p>
  </div>
  <p
    data-testid="number-of-accounts"
  >
    Number of accounts: 
    70
  </p>
</div>
`;
